src/libs/tools/src/plugindatabase.cpp

```diff
@@ -367,5 +364,39 @@ std::string MockPluginDatabase::lookupInfo (PluginSpec const & spec, std::string
 
 	return "";
 }
+
+/**
+* @brief reference checkconf implementation for the unit tests.
+* 
+* Checks for the key "a" to be set to "abc" and fails if it can not find it.
+* Checks for the key "b" and appends it to the configuration if it can not find it.
+*/
+static int checkconfMock (ckdb::Key * errorKey, ckdb::KeySet * conf)
+{
+	ckdb::Key * k;
+
+	k = ckdb::ksLookupByName (conf, "user/a", 0);
+	if (!k || strncmp (ckdb::keyString (k), "abc", 3))
+	{
+		return -1; // invalid config, could not fix
+	}
+
+	k = ckdb::ksLookupByName (conf, "user/b", 0);
+	if (!k)
+	{
+		ckdb::ksAppendKey (conf, ckdb::keyNew ("user/b", 0));
+		return 1; // changed config, OK
+	}
+	return 0; // did not change config, OK
+}
+
+PluginDatabase::func_t MockPluginDatabase::getSymbol (PluginSpec const & spec, std::string const & which) const
+{
+	if (which.find ("checkconf") != std::string::npos && spec.getFullName ().find ("checkconf") != std::string::npos)
```

Comment:
I am not sure what this code should do.
First part: Why not simply comparing `which == "checkconf`? Second part: Why should the name of spec contain checkconf? Why not adding the mock function in every test case as needed?